src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+public class FigureSupplier {
+
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        Figure[] figures = new Figure[5];
+        figures[0] = new Circle(colorSupplier.getRandomColor(),Math.random() * 100);
+        figures[1] = new Rectangle(colorSupplier.getRandomColor(),Math.random() * 100,
+                Math.random() * 100);
+        figures[2] = new IsoscelesTrapezoid(colorSupplier.getRandomColor(),Math.random() * 100,
+                Math.random() * 100, Math.random() * 100);
+        figures[3] = new RightTriangle(colorSupplier.getRandomColor(), Math.random() * 100,
+                Math.random() * 100);
+        figures[4] = new Square(colorSupplier.getRandomColor(), Math.random() * 100);
+        int count = (int)(Math.random() * 4);
```

Comment:
Notice that you create multiple object even though you return only one - please use a proper instructions to build only the requested figure. How about assigning each figure to a number 0-4 and use switch?